Allow configuring which scheme (https or http) to consider when generating the
SAML Consumer Service Assertions (i.e. the URL where to return after the IdP (Keycloak) login).

diff --git /usr/share/univention-management-console/saml/sp.py /usr/share/univention-management-console/saml/sp.py
index 8164554..c9dd8e8 100644
--- /usr/share/univention-management-console/saml/sp.py
+++ /usr/share/univention-management-console/saml/sp.py
@@ -1,5 +1,6 @@
 # http://pysaml2.readthedocs.org/en/latest/howto/config.html
 import glob
+import os
 import os.path
 from tempfile import NamedTemporaryFile

@@ -55,9 +56,10 @@ else:
     addresses = [fqdn]
     addresses.extend([y['address'] for x, y in i.all_interfaces if y and y.get('address')])

-bases = ['%s://%s/univention/saml' % (scheme, addr) for addr in addresses for scheme in ('https', 'http')]
+schemes = tuple(scheme.strip() for scheme in os.getenv('SAML_SCHEMES', 'https, http').split(','))
+bases = ['%s://%s/univention/saml' % (scheme, addr) for addr in addresses for scheme in schemes]
 CONFIG = {
-    "entityid": "https://%s/univention/saml/metadata" % (fqdn,),
+    "entityid": "%s://%s/univention/saml/metadata" % (schemes[0], fqdn,),
     "name_form": NAME_FORMAT_URI,
     "description": "Univention Management Console at %s" % (fqdn,),
     "service": {
